site/source/docs/optimizing/Optimizing-Code.rst

```diff
@@ -148,30 +148,21 @@ Other optimization issues
 C++ exceptions
 --------------
 
-C++ exceptions are turned off by default in ``-O1`` (and above). This prevents the generation of ``try-catch`` blocks, which lets the code run much faster, and also makes the code smaller. 
+Catching C++ exceptions is turned off by default in ``-O1`` (and above). This prevents the generation of ``try-catch`` blocks, which lets the code run much faster, and also makes the code smaller. 
 
 To re-enable exceptions in optimized code, run *emcc* with ``-s DISABLE_EXCEPTION_CATCHING=0`` (see `src/settings.js <https://github.com/kripken/emscripten/blob/master/src/settings.js>`_).
 
+.. note:: Removing catching of exceptions helps a lot with speed, but for code size you should build with `-fno-exceptions` to really get rid of all exceptions code overhead, as it may contain thrown exceptions that are never caught (e.g. just using std::vector can have that). `-fno-rtti` may help as well.
```

Comment:
I'd probably be more specific that this is extra size overhead; in other words maybe we could organize it this way:
1. By default catch blocks are removed and throws terminate the program.
2. Enabling exception catching has a big speed and size overhead compared to the default, but lets you actually use exceptions
3. Using `-fno-exceptions` has a size advantage compared to the default, but exception-using code will fail to compile.
4. Using `-fno-rtti` may have a further size advantage (but of course then you can't use RTTI for e.g. `dynamic_cast`).

Reply:
Usually there are practical challenges (a lot of intrusive codechanges) in using -fno-exceptions. 
Will a manual DCE (dead code elimination) or commenting of source code reachable from catch block (say ensuring all catch blocks are empty in application source code and libcxx source code) help in achieving the same size optimization as proposed in point 3 above ?
It is possible to use -fno-rtti without using -fno-exception to strip out long symbol names from mem file and use -s DISABLE_EXCEPTION_CATCHING=1 at link time.

Reply:
1. Yeah, that's why we haven't made `-fno-exceptions` the default.
2. I think not, because with `-fexceptions` even code which has no try/catch blocks has to use invoke instead of call for functions which may propagate exceptions, and this has some cost (e.g. invoke terminates the basic block and so inhibits some other optimizations).
3. True, but probably not relevant to this particular part of the doc. Do we have a section on general code size reduction?
4. I'm not exactly sure what you mean? but if I understand you correctly, I don't think doing that would be much better than just running lower-invoke as we do now.

Reply:
I rewrote that section a little after this feedback.